scripts/zones/Windurst_Waters_[S]/npcs/Hampu-Kampu.lua

```diff
@@ -0,0 +1,54 @@
+-----------------------------------
+-- Area: Windurst Waters (S)
+--  NPC: Hampu-Kampu
+-- Type: Quest NPC
+-----------------------------------
+local ID = require("scripts/zones/Windurst_Waters_[S]/IDs")
+require("scripts/globals/quests");
+require("scripts/globals/keyitems");
+
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+    local sayItWithAHandbag = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+
+    if (sayItWithAHandbag == QUEST_COMPLETED) then
+        player:startEvent(175)
+    elseif (player:hasKeyItem(dsp.ki.REPAIRED_HANDBAG) and sayItWithAHandbagCS == 4) then
+        player:startEvent(174)
+        print("repaired bag and 4")
+    elseif (player:hasKeyItem(dsp.ki.TORN_PATCHES_OF_LEATHER) or sayItWithAHandbagCS == 3) then
+        player:startEvent(173)
+        print(player:hasKeyItem(dsp.ki.REPAIRED_HANDBAG))
+    elseif (sayItWithAHandbag == QUEST_ACCEPTED and sayItWithAHandbagCS == 1) then
+        player:startEvent(172)
+    elseif (sayItWithAHandbag == QUEST_ACCEPTED) then
+        player:startEvent(170)
+    elseif (sayItWithAHandbag == QUEST_AVAILABLE) then
+        player:startEvent(169)
+        player:setVar("sayItWithAHandbagCS",0)
+    end
+end
+
+function onEventUpdate(player,csid,option)
+end
+
+function onEventFinish(player,csid,option)
+    if (csid == 174) then -- Option doesn't matter as NPC will take key item if yes or no
+        if (player:getFreeSlotsCount() < 1) then
+            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED,19110);
+        else
+            player:delKeyItem(dsp.ki.REPAIRED_HANDBAG)
+            player:addItem(19110)
+            player:completeQuest(CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG)
```

Comment:
You can use [npcUtil.completeQuest](https://github.com/DarkstarProject/darkstar/blob/master/scripts/globals/npc_util.lua#L268-L285) to reduce this down. The first thing it tries to do is attempt to give items, and if it can't it'll return false. So you can just check against it; if the player can receive the item, well, you've completed the quest. Make sure you only delete the KI if it returned true.

Also, you forgot to clear out your stage var~!

```
if npcUtil.completeQuest(player, CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG, {item=19110}) then
    player:delKeyItem(dsp.ki.REPAIRED_HANDBAG)
    player:setVar("sayItWithAHandbagCS",0)
end
```

(`npcUtil.completeQuest` can take a var in its table too to clear out, but the line might get a little long)

Reply:
@ibm2431  OMG I am loving npcUtil ❤❤❤
Unfortunately I cannot set the Var to  0 cause it has an optional CS afterwards
I guess I can set it to 0 but then I would still have to change it to 5, but wouldn't that be sort of confusing for reading? If it didn't have an after cutscene I would totally do it to make it simpler. What do you think?

Reply:
Changed var to 0 as using a different one to track Bonus CS